src/sprite_battler.cpp

```diff
@@ -264,19 +264,20 @@ bool Sprite_Battler::IsIdling() {
 }
 
 int Sprite_Battler::GetWidth() const {
-	if (animation) {
-		return animation->GetWidth();
-	}
+	if (animation) return 48;
```

Comment:
Why assume 48? If we ever support custom large animation formats this would break.

Why doesn't `animation->GetWidth()` work here?

Reply:
Because `animation->GetWidth()` returns 0 for some reason. Battle animations normally have a width and height of 96 but for some reason 48 is needed to place the actors correctly.

Reply:
That's because the underlying graphical resources haven't been loaded yet, and so we don't . Take a look at `Scene_Battle_Rpg2k3` and the `Start()` and `Start2()` functions. Notice how we use this `AsyncNext` mechanism for `Start2`.

This exists because anytime some logic depends on loading resources, we have to wait for them to be loaded. We have a asynchronous IO system here because of platforms like emscripten, where data can take a couple frames to load. 

Basically what happens here is:

1. Screen Erased
2. Start() called - does setup logic
3. Start() registers Start2() to be called when all pending async IO is done
4. System waits for async files to be loaded
5. Start2() is called
6. TransitionIn and begin the scene.

All of the battler placement logic is done in Start2() for this reason, as it depends on battler sprites. Also notice at the end how we reset the battler sprite Z files.

TLDR; Do your animation initialization login in `Start2()`

Reply:
>  for some reason 48 is needed to place the actors correctly.

It's the half width. 96 / 2 = 48 

The placement logic works the same way, RPG_RT hard codes 24. We do battler sprite width (48) / 2 = 24